stratum/hal/lib/tdi/dpdk/dpdk_hal.cc

```diff
@@ -207,14 +210,41 @@ ::util::Status DpdkHal::Run() {
   const std::vector<std::string> external_stratum_urls =
       absl::StrSplit(FLAGS_external_stratum_urls, ',');
   {
+    std::string log_output_str;
     ::grpc::ServerBuilder builder;
     SetGrpcServerKeepAliveArgs(&builder);
 
-    builder.AddListeningPort(FLAGS_local_stratum_url,
-                             ::grpc::InsecureServerCredentials());
-
-    for (const auto& url : external_stratum_urls) {
-      builder.AddListeningPort(url, ::grpc::InsecureServerCredentials());
+    if (FLAGS_grpc_open_insecure_ports) {
+      // User has chosen to open insecure ports
+      LOG(WARNING) << "Warning: Flag set to open gRPC insecure ports";
+      log_output_str = "[insecure mode] ";
+      builder.AddListeningPort(FLAGS_local_stratum_url,
+                              ::grpc::InsecureServerCredentials());
+
+      for (const auto& url : external_stratum_urls) {
+        builder.AddListeningPort(url, ::grpc::InsecureServerCredentials());
+      }
+    } else {
+      log_output_str = "[secure mode] ";
+      auto credentials_manager = stratum::CredentialsManager::CreateInstance(true);
+      if (!credentials_manager.ok()) {
+        LOG(ERROR) << "Credentials Manager initialization failed. Unable to open ports for gRPC";
+      } else {
+        auto resp = credentials_manager.ConsumeValueOrDie();
+        auto server_credentials = resp.get()->GenerateExternalFacingServerCredentials();
+        if (server_credentials == nullptr) {
+          return MAKE_ERROR(ERR_INTERNAL)
```

Comment:
As we are not falling back to non-secure mode in this scenario, better to terminate the process along with the required log as without the P4 Runtime interface nothing can be done.

Reply:
I saw that it doesnt quit. Talking to @vsureshkumarp we want the program to exit if server credentials are not loaded.